okhttp/src/main/java/io/grpc/transport/okhttp/AsyncFrameWriter.java

```diff
@@ -45,20 +46,26 @@
 import java.io.IOException;
 import java.util.List;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Executor;
 
 class AsyncFrameWriter implements FrameWriter {
-  private final FrameWriter frameWriter;
-  private final Executor executor;
+  private volatile FrameWriter frameWriter;
```

Comment:
Although you are correct in wanting this value to be volatile, but I think we may not want it to be. We don't actually need it volatile, because we all references to it are on the same thread (via `executor`). And volatile is more noticeable on ARM (unlike x86). Can we just specify that `setFrameWriter()` must be called from `executor`?